src/device/usbd.c

```diff
@@ -309,11 +319,18 @@ bool tud_init (void)
   _usbd_q = osal_queue_create(&_usbd_qdef);
   TU_ASSERT(_usbd_q != NULL);
 
+  // Get application driver if available
+  if ( usbd_app_driver_get_cb )
+  {
+    _app_driver = usbd_app_driver_get_cb(&_app_driver_count);
```

Comment:
app driver callback when init.

Reply:
Does this callback apply both to host and device drivers? (If not, suggest indicating which ones in the name)

Reply:
yeah, host if supported will have its own callbacks :). Host and device stack are independent to each other. Only sharing common usb definitions and types